### ISSUE TITLE
KeepCommentsRegex not working for knockoutJS comments in 1.16.1

### ISSUE THREAD

**Issue by mattscotty** [trullock/NUglify#282]
I'm trying to keep knockoutJS comments but they are still being stripped out.

Following [this code](https://github.com/trullock/NUglify/blob/master/src/NUglify/Html/HtmlSettings.cs) I tried adding the knockout regex (I suspect its in by default but isn't working for me);

```
NUglify.Html.HtmlSettings htmlSettings = new NUglify.Html.HtmlSettings();
htmlSettings.KeepCommentsRegex.Add(new Regex(@"^/?ko(?:[\s\-]|$)")); // Keep knockout comments

return Uglify.Html(PathToMyHTML, htmlSettings).Code;
```

An example of the code being stripped out is;

```
<!-- ko if: $root.Videos().length > 0 -->
<!-- /ko -->
```

**Comment by trullock** [comment]
Not had a proper look yet, but theres a test here for knockout: https://github.com/trullock/NUglify/blob/master/src/NUglify.Tests/Html/TestComments.cs

Whats different between that and what youre doing?

**Comment by mattscotty** [comment]
The link helped a lot.
It turns out we have spaces before and after the statements to help with readability.
Removing the spaces fixes the issue.

As we have a lot of these in our code, all with spaces (as shown in the [knockout examples](https://knockoutjs.com/documentation/if-binding.html#note-using-if-and-ifnot-without-a-container-element) which show spaces), would it be possible to modify the base regex to allow spaces?
If its like that in the knockout docs, I suspect other people will be caught out by this.

To confirm;
*Works*
```
<!--ko if: $root.Videos().length > 0-->
<!--/ko-->
```

*Doesn't work*
```
<!-- ko if: $root.Videos().length > 0 -->
<!-- /ko -->
```

Thanks

**Comment by trullock** [comment]
Aha, yeah it shouldn't be whitespace sensitive, should be an easy fix!

**Comment by mattscotty** [comment]
One thing to add, I'm not sure if its worth it, but you could minify the surrounding whitespace around the KO statement's if present as I presume its not needed and its just a read-ability thing.

**Comment by trullock** [comment]
> One thing to add, I'm not sure if its worth it, but you could minify the surrounding whitespace around the KO statement's if present as I presume its not needed and its just a read-ability thing.

Do you want to break this out into a separate issue, as its not KO specific

**Comment by trullock** [comment]
Whitespace issue fixed in 1.16.4
